src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.impl.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> operationMap) {
+        this.operationMap = operationMap;
+    }
+
+    @Override
+    public OperationHandler getService(FruitTransaction.Operation operation) {
+        return operationMap.get(operation);

```

Consider adding a check to handle cases where the operation is not found in the `operationMap`. You could throw an exception or return a default handler to ensure robustness.